### ISSUE TITLE
Segfault on moving node

### ISSUE THREAD

**Issue by RM87** [openstreetmap/merkaartor#64]
Loaded a saved file, created a node and moved it  for 2-3 times.

0x0000000000493f18 in MemoryBackend::purge (this=0xa8e5a0 <g_backend>)
    at Backend/MemoryBackend.cpp:394
394	        delete *(it++);

Backtrace:
```
#0  0x0000000000493f18 in MemoryBackend::purge (this=0xa8e5a0 <g_backend>)
    at Backend/MemoryBackend.cpp:394
#1  0x0000000000493fc9 in MemoryBackend::resumeDeletes (this=0xa8e5a0 <g_backend>)
    at Backend/MemoryBackend.cpp:406
#2  0x00000000004ff53d in RenderTile::operator() (this=0x2776098, theTile=...)
    at Layers/OsmRenderLayer.cpp:109
#3  0x00000000005018d1 in QtConcurrent::MapKernel<QList<QPoint>::iterator, RenderTile>::runIteration (this=0x2776060, it=...)
    at /usr/include/qt4/QtCore/qtconcurrentmapkernel.h:73
#4  0x0000000000501955 in QtConcurrent::MapKernel<QList<QPoint>::iterator, RenderTile>::runIterations (this=0x2776060, sequenceBeginIterator=..., beginIndex=28, 
    endIndex=29) at /usr/include/qt4/QtCore/qtconcurrentmapkernel.h:82
#5  0x0000000000501de9 in QtConcurrent::IterateKernel<QList<QPoint>::iterator, void>::forThreadFunction (this=0x2776060)
    at /usr/include/qt4/QtCore/qtconcurrentiteratekernel.h:263
#6  0x0000000000501ad6 in QtConcurrent::IterateKernel<QList<QPoint>::iterator, void>::threadFunction (this=0x2776060)
    at /usr/include/qt4/QtCore/qtconcurrentiteratekernel.h:225
#7  0x00007ffff2ac6bed in QtConcurrent::ThreadEngineBase::run (this=0x27760a0)
    at concurrent/qtconcurrentthreadengine.cpp:264
#8  0x00007ffff2ac8be0 in QThreadPoolThread::run (this=0x7fffbc00b430)
    at concurrent/qthreadpool.cpp:108
#9  0x00007ffff2ad586f in QThreadPrivate::start (arg=0x7fffbc00b430)
    at thread/qthread_unix.cpp:352
#10 0x00007ffff28440a4 in start_thread (arg=0x7fffc0933700) at pthread_create.c:309
#11 0x00007ffff1d5607d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
```
core: https://app.box.com/s/x615xhpck6eeshebeawm4z8oxiab775n
savefile  before creating that node: https://app.box.com/s/uo6fd98f3xgdqaz6x65ihjh4lfw8jlz0

node location:
 ![image](https://cloud.githubusercontent.com/assets/859636/9314222/3e00d3f0-4530-11e5-9a8a-d9dd4ff9429d.png)

**Comment by RM87** [comment]
Re-creatable on my machine by moving newly created node quickly several times by a small fraction

**Comment by Krakonos** [comment]
It might be fixed in d424fa8114909a872a9ec8c68978588d6e4512b7, could you confirm it?

**Comment by RM87** [comment]
At least that one I was not able to reproduce anymore.
